Algorithmia/CLI.py

```diff
@@ -0,0 +1,331 @@
+import Algorithmia
+import os
+from Algorithmia.algo_response import AlgoResponse
+import json, re, requests, six
+import toml
+
+class CLI():
+    def __init__(self):
+        self.client = Algorithmia.client()
+        # algo auth
+    def auth(self, apikey, apiaddress, profile):
+
+        #store api key in local config file and read from it each time a client needs to be created
+        key = self.getconfigfile()
+        config = toml.load(key)
+
+        if('profiles' in config.keys()):
+            if(profile in config['profiles'].keys()):
+                config['profiles'][profile]['api_key'] = apikey
+                config['profiles'][profile]['api_server'] = apiaddress
+            else:
+                config['profiles'][profile] = {'api_key':apikey,'api_server':apiaddress}
+        else:
+            config['profiles'] = {profile:{'api_key':apikey,'api_server':apiaddress}}
+
+        key = open(key, "w")
+
+        toml.dump(config,key)
+        key.close()
+
+        self.ls(path = None,client = Algorithmia.client(self.getAPIkey(profile)))
+
+    # algo run <algo> <args..>    run the the specified algo
+    def runalgo(self, options, client):
+        algo_input = None
+
+        algo = client.algo(options.algo)
+
+        result = None
+
+        algo.set_options(timeout=options.timeout, stdout=options.debug)
+
+
+
+        #handle input type flags
+        if(options.data != None):
+            #data
+            algo_input = options.data
+
+            result = algo.pipe(algo_input)
+
+        elif(options.text != None):
+            #text
+            algo_input = options.text
+            key = self.getAPIkey(options.profile)
+            result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input.encode('utf-8'),
```

Comment:
For these similar function calls, I may prepare the parameters that I would reuse beforehand, like the `client.apiAddress + algo.url` and then use this variable in every function call. The same could go for the header, by preparing a dictionary for the keys that stay the same in every function call and enriching that dictionary with the changing key-value parameters when I needed a different version of it. 

This would help when we want to refactor the code in the future, like to add something else to the header or change the URL, etc... . We wouldn't have to do the same change for every function call but only do it in as little places as possible. 

Also, it helps the reader of the code to understand what's same between the function calls and what's different.